### PR TITLE
Add sort - Closes Issue #2

### PR DESCRIPTION
**Pull request by erichoog**
Hello, I hope this finds you well.

I've implemented a SortHelper class for issue #2 I have done my best to keep to your coding style.

I'd love to hear any feedback you might have on this PR.

One caveat is that the + in the query string is encoded to a blank space so `?sort=location,+age` will be the same as `?sort=location, age` (notice the space).

Would love to hear some feedback.  Thanks!

### CODE REVIEW

**Comment by ttu** [comment]
Thanks for the feature! Will check the code in few days, but looks good 👍 

You can also add a new line to the CHANGELOG.md, e.g. Add query support for sorting

File: FakeServer/Common/SortHelper.cs [hunk 1]
```diff
@@ -0,0 +1,73 @@
+﻿using System;
```
Comment (ttu) [inline FakeServer/Common/SortHelper.cs]: I prefer handling first the first item and then looping through the rest, instead of having a conditional check inside a loop. If conditional check is used I prefer to use check with boolean isFirst.

Rest of the methods can be "simplified" a little.

```cs
using System.Collections.Generic;
using System.Dynamic;
using System.Linq;

namespace FakeServer.Common
{
    public static class SortHelper
    {
        public static IEnumerable<dynamic> SortFields(IEnumerable<dynamic> results, IEnumerable<string> sortFields)
        {
            if (sortFields.Where(e => !string.IsNullOrEmpty(e)).Any() == false)
                return results;

            var isSortDescending = IsSortDescending(sortFields.First());
            var fieldName = RemoveSortDirection(sortFields.First());

            var sortedResults = GetFirstSortResults(results, fieldName, isSortDescending);

            sortFields.Skip(1).ToList().ForEach(x =>
            {
                isSortDescending = IsSortDescending(x);
                fieldName = RemoveSortDirection(x);

                sortedResults = GetRemainingSortResults(sortedResults, fieldName, isSortDescending);
            });

            return sortedResults.AsEnumerable();
        }

        private static IOrderedEnumerable<dynamic> GetFirstSortResults(IEnumerable<dynamic> results, string sortField, bool isSortDescending)
        {
            if (isSortDescending)
                return results.OrderByDescending(x => ParseField(x as ExpandoObject, sortField));
            else
                return results.OrderBy(x => ParseField(x as ExpandoObject, sortField));
        }

        private static IOrderedEnumerable<dynamic> GetRemainingSortResults(IOrderedEnumerable<dynamic> results, string sortField, bool isSortDescending)
        {
            if (isSortDescending)
                return results.ThenByDescending(x => ParseField(x as ExpandoObject, sortField));
            else
                return results.ThenBy(x => ParseField(x as ExpandoObject, sortField));
        }

        private static dynamic ParseField(ExpandoObject s, string field) => (s as IDictionary<string, object>)[field];

        private static string RemoveSortDirection(string sortField) => sortField.Replace("+", "").Replace("-", "").Trim();

        private static bool IsSortDescending(string sortField) => !(char.IsWhiteSpace(sortField[0]) || sortField.Contains("+"));
    }
}
````

File: FakeServer.Test/SortHelperTests.cs [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+using System.Dynamic;
+using System.Linq;
+using System.Threading;
+using Xunit;
+
+namespace FakeServer.Test
+{
+    public class SortHelperTests
+    {
+        [Fact]
```
Comment (ttu) [inline FakeServer.Test/SortHelperTests.cs]: You can add test for a case when sortFields has an empty string

```cs
[Fact]
        public void SortFields_EmptySortArray()
        {
            var result = SortHelper.SortFields(new dynamic[] { }, new string[] { });
            Assert.NotNull(result);

            result = SortHelper.SortFields(new dynamic[] { }, new[] { "" });
            Assert.NotNull(result);
        }
```

Comment (ttu) [reply FakeServer.Test/SortHelperTests.cs]: Also remove unnecessary usings statements and empty lines. I use http://www.codemaid.net/ for cleaning.

Comment (erichoog) [reply FakeServer.Test/SortHelperTests.cs]: Do you know of any good similar extensions for VS Code?

Comment (erichoog) [reply FakeServer.Test/SortHelperTests.cs]: Thanks very much for the feedback.  I think I have completed all of the changes you've requested other than the removing the unnecessary using statements and empty lines.  I'm currently using VS Code so CodeMaid isn't supported.  I'll continue to look for a good extension to complete that.

**Comment by ttu** [comment]
Thanks for the update!

**Comment by erichoog** [comment]
My pleasure and thank you for the great feedback!
